src/domain/products/index.js

```diff
@@ -68,17 +78,98 @@ const ProductIndex = () => {
   } = useMedusa("products", {
     search: {
       ...filtersOnLoad,
-      fields: "id,title,thumbnail",
-      expand: "variants,variants.prices,collection",
+      ...defaultQueryProps,
     },
   })
+
+  const tagsPromise = Medusa.products.listTagsByUsage()
+
   const [query, setQuery] = useState("")
   const [limit, setLimit] = useState(filtersOnLoad.limit || 20)
   const [offset, setOffset] = useState(filtersOnLoad.offset || 0)
   const [storeCurrencies, setStoreCurrencies] = useState([])
+  const [collectionsList, setCollectionsList] = useState([])
   const [selectedProduct, setSelectedProduct] = useState()
   const [copyingProduct, setCopyingProduct] = useState(false)
 
+  const [statusFilter, setStatusFilter] = useState({
+    open: false,
+    filter: null,
+  })
+  const [collectionFilter, setCollectionFilter] = useState({
+    open: false,
+    filter: null,
+  })
+  const [tagsFilter, setTagsFilter] = useState([])
+  const [filterTags, setFilterTags] = useState(false)
+
+  const resetFilters = () => {
+    setStatusFilter({
+      open: false,
+      filter: null,
+    })
+    setCollectionFilter({
+      open: false,
+      filter: null,
+    })
+    setTagsFilter([])
+    setFilterTags(false)
+  }
+
+  const submit = async () => {
+    const collectionIds = collectionFilter.filter
+      ? collectionFilter.filter
+          .split(",")
+          .map(cf => collections.find(c => c.title === cf)?.id)
+          .join(",")
+      : null
+
+    const tagsResponse = await tagsPromise
+    const tagIds = tagsFilter
+      .map(tag => tag.trim())
+      .map(tag => tagsResponse.data.tags.find(t => t.value === tag)?.id)
+      .join(",")
+
+    const urlObject = {
+      "status[]": statusFilter.open ? statusFilter.filter : null,
+      "collection_id[]": collectionFilter.open ? collectionIds : null,
+      "tags[]": filterTags && tagIds?.length > 0 ? tagIds : null,
+    }
+    const url = { ...removeNullish(urlObject) }
+
+    replaceQueryString(url)
+  }
+
+  const replaceQueryString = queryObject => {
+    let searchObject = {
+      ...queryObject,
+      ...defaultQueryProps,
+    }
+
+    if (_.entries(queryObject).length === 0) {
+      resetFilters()
+      window.history.replaceState({}, "", "/a/products")
+      refresh({ search: { ...defaultQueryProps } })
+    } else {
+      if (!searchObject.offset) {
+        searchObject.offset = 0
+      }
+
+      if (!searchObject.limit) {
+        searchObject.limit = 20
+      }
+
+      const query = qs.stringify(queryObject)
+      window.history.replaceState(`/a/products`, "", `${`?${query}`}`)
+      refresh({ search: { ...searchObject } })
+    }
+  }
+
+  const clearFilters = () => {
```

Comment:
Upon clearing filters, we should use the default offset and limit. Currently, we fetch all products.